File: santhuongmai/santhuongmai/src/main/java/com/example/santhuongmai/service/VoucherService.java
package com.example.santhuongmai.service;

import java.util.List;


import com.example.santhuongmai.entity.Voucher;
import com.example.santhuongmai.model.request.CreateVoucherRequest;

public interface VoucherService {
	
	List<Voucher> getListVoucher(long id);
	Voucher createVoucher(CreateVoucherRequest request);
	Voucher updateVoucher(long id, CreateVoucherRequest request);
	void deleteVoucher(long id);
	List<Voucher> getList();
	void enableVoucher(long id);
	Voucher updateVouchercount(long id,CreateVoucherRequest request);
}

File: Frontendv1/src/app/_service/voucher.service.ts
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { SERVER_DOMAIN } from './domain.service';

// const VOUCHER_API = "http://localhost:8080/api/voucher/";
const VOUCHER_API = SERVER_DOMAIN + "/api/voucher/";
const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable({
  providedIn: 'root'
})
export class VoucherService {
  // products : any[] =[];
 
  constructor(private http:HttpClient) { }

  getList():Observable<any>{
    return this.http.get(VOUCHER_API,httpOptions);
  }


  getListVoucher(id: number):Observable<any>{
    return this.http.get(VOUCHER_API + 'uservoucher/' + id,httpOptions);
  }
  enableVoucher(id: number){
    return this.http.put(VOUCHER_API + 'enable/'+ id,httpOptions);
  }
  

  

  createVoucher(id:number,name:string,count:number,money: string):Observable<any>{
    return this.http.post(VOUCHER_API +'create',{id,name,count,money},httpOptions);
  }

  updateVoucher(id:number,name:string,count:number,money: string):Observable<any>{
    return this.http.put(VOUCHER_API + 'update/'+id,{name,count,money},httpOptions);
  }

  updateVouchercount(id:number,count:number):Observable<any>{
    return this.http.put(VOUCHER_API + 'updatecount/'+id,{count},httpOptions);
  }

  deleteVoucher(id:number):Observable<any>{
    return this.http.delete(VOUCHER_API + 'delete/' + id,httpOptions);
  }



}
